.github/workflows/build-linux.yml

```diff
@@ -59,46 +69,155 @@ jobs:
+          - { id: ubuntu-22.04, name: jammy }
+        compiler:
+          - 'clang-9'
+          - 'clang-10'
+          - 'clang-11'
+          - 'clang-12'
+          - 'gcc-7'
+          - 'gcc-8'
+          - 'gcc-9'
+          - 'gcc-10'
+          - 'gcc-11'
+        cpp_std:
+          - 'c++11'
+          - 'c++14'
+          - 'c++17'
+        exclude:
+          # These versions are not supported by the toolchain PPA
+          - { os: { id: ubuntu-22.04 }, compiler: gcc-7 }
+          - { os: { id: ubuntu-22.04 }, compiler: gcc-8 }
+          - { os: { id: ubuntu-22.04 }, compiler: clang-9 }
+          - { os: { id: ubuntu-22.04 }, compiler: clang-10 }
+      fail-fast: false
+    steps:
+      - name: Runtime environment
+        shell: bash
+        env:
+          WORKSPACE: ${{ github.workspace }}
+        run: |
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+          echo "GITHUB_WORKSPACE=`pwd`" >> $GITHUB_ENV
+      - name: Coverage environment
+        run: |
+          echo "BUILD_OPTS=--buildtype=debug" >> $GITHUB_ENV
+      - name: Setup GCC
+        if: startsWith(matrix.compiler, 'gcc')
+        shell: bash
+        run: |
+          CXX=${CC/#gcc/g++}
+          sudo apt-add-repository ppa:ubuntu-toolchain-r/test
+          sudo apt-get update
+          sudo apt-get install $CC $CXX gcovr
+          echo "CC=$CC" >> $GITHUB_ENV
+          echo "CXX=$CXX" >> $GITHUB_ENV
+          echo "GCOV=${CC/#gcc/gcov}" >> $GITHUB_ENV
+        env:
+          CC: ${{ matrix.compiler }}
+      - name: Setup Clang
+        if: startsWith(matrix.compiler, 'clang')
+        shell: bash
+        run: |
+          wget https://apt.llvm.org/llvm-snapshot.gpg.key
+          sudo apt-key add llvm-snapshot.gpg.key
+          rm llvm-snapshot.gpg.key
+          sudo apt-add-repository "deb https://apt.llvm.org/${{ matrix.os.name }}/ llvm-toolchain-${{ matrix.os.name }} main"
+          sudo apt-get update
+          CXX=${CC/#clang/clang++}
+          sudo apt-get install $CC $CXX gcovr
+          echo "CC=$CC" >> $GITHUB_ENV
+          echo "CXX=$CXX" >> $GITHUB_ENV
+          echo "GCOV=/usr/lib/${CC/#clang/llvm}/bin/llvm-cov gcov" >> $GITHUB_ENV
+        env:
+          CC: ${{ matrix.compiler }}
+        working-directory: ${{ runner.temp }}
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          lfs: true
+          submodules: true
+      - name: Setup Meson + Ninja
+        shell: bash
+        run: |
+          sudo pip3 install --upgrade pip setuptools wheel
+          pip3 install --user meson ninja
+        working-directory: ${{ runner.temp }}
+      - name: Version tools
+        shell: bash
+        run: |
+          $CC --version
+          $CXX --version
+          $GCOV --version
+          meson --version
+          ninja --version
+      - name: Configure
+        shell: bash
+        run: meson build --prefix=$HOME/.local -Db_coverage=true -Dcpp_std=${{ matrix.cpp_std }} $BUILD_OPTS
```

Comment:
given BUILD_OPTS will always be set to`--buildtype=debug` here, we can probably dispense with it and the `Coverage environment` step

Reply:
As discussed on private, we'll move the `-Db_coverage=1` line here too, and make the step more defensive-proof.